src/core/utils/request/getRequestCookies.ts

```diff
@@ -66,7 +66,10 @@ export function getAllRequestCookies(request: Request): Record<string, string> {
    * is pure-er.
    */
   for (const [name, value] of Object.entries(forwardedCookies)) {
-    request.headers.append('cookie', `${name}=${value}`)
+    request.headers.append('cookie', cookieUtils.serialize(name, value))
+    if (value.indexOf('%') !== -1) {
```

Comment:
I don't think particular substring detection in the cookie value is a reliable strategy here. 

- Can we use `decode` option (or similar) in the `cookieUtils.serialize()` instead? Why not?

Reply:
In fact, `cookieUtils.serialize()` use this detection ([cookie/index-esm.js](https://github.com/bundled-es-modules/cookie/blob/db6d3d885758a239e7df126c8e990f55fec188fa/index-esm.js#L150C1-L152C6)) internally. But I'll revert my last commit cause my previous commits already fulfilled my needs.

Reply:
I didn't know that! Thanks for reverting to the previous state, it's much more straightforward now. Giving this a look...